### DISCUSSION TITLE
I don't understand how to MIDI learn a keyboard in the new firmware

### DISCUSSION THREAD

**Discussion by miaouxmiaoux** [SynthstromAudible/DelugeFirmware#2284] [Q&A]
### Please describe the problem:

Tested with the official firmware, so I know MIDI routing is correct. I've watched Ron C's video on it and in MIDI Follow -> Follow -> Channel A his screen has "DIN Ports" on the second row while my Deluge says "Device Unassigned". Holding learn and pressing a key on the keyboard I want to use, as he does, does nothing. The old workflow - holding learn and pressing the 'Audition / Section' row - does nothing. What am I doing wrong?

### What is the expected behavior?

I want to MIDI learn a keyboard

### What hardware did you reproduce it with?

OLED

### What firmware did you reproduce it with?

Release 1.1/Beethoven

**Comment by sapphire-arches** [comment]
You should still be able to learn midi devices to clips the same way. To learn a MIDI follow channel, hold learn/input while in that menu and press a key on the controller.

**Comment by miaouxmiaoux** [comment]
Thanks, neither of these are working. It must be a problem with my setup, but it's odd because MIDI learn works with the official firmware but not Beethoven. So all the routing is correct and the Deluge is receiving MIDI from the USB port. But as soon as I load Beethoven the Deluge no longer receives MIDI over USB.

**Comment by m-m-adams** [comment]
Are you on a mac or iPad? If so it's an Apple bug and you need to reset your MIDI settings to fix it, then connect the deluge running community firmware.

Official will still work after this

**Comment by miaouxmiaoux** [comment]
Thanks! I'm on a Mac. The Deluge and all MIDI devices are connected via an iConnectivity mioXL router.

I'll give this a go, what exactly do you mean by 'reset MIDI settings'?

**Comment by m-m-adams** [comment]
Oh if you're on a mioXL remove the deluge from USB host reservation, restart the mioXL, then re add deluge port 1

<img width="1002" alt="image" src="https://github.com/SynthstromAudible/DelugeFirmware/assets/52469396/28ebc6c6-35c7-42b8-94d0-d99885ce3a07">

should look something like this afterwards. We added extra virtual ports but the mioXL won't see them yet


If you use USB right to the mac then you basically do the same thing in MIDI Studio - delete the deluge and unplug it, rescan midi, then plug the deluge back in

**Reply by miaouxmiaoux** [reply]
Thanks, I've tried this and it still isn't working. Power cycled the mioXL, removed the Deluge from USB host reservation, deleted the mioXL from MIDI studio and plugged everything back in - still nothing.

I'd like to reiterate that the MIDI learn function is working perfectly with this setup and the official OS. So Beethoven is the issue I think.

**Comment by duncanrmi** [comment]
I need to check what OS I'm running, but I have two deluges & each has an akai midimix on its usb port, with the deluge running from a 9V supply. tried to use one of the midimix devices this morning, won't respond. thought I'd forgotten how to use 'learn'... same akai device is fine on blackbox/bluebox.
it's like usb-midi has just stopped working.
